src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -123,6 +125,31 @@ public string HostId
             }
         }
 
+        /// <summary>
+        /// Gets or sets the host instance ID. If not explicitly set, the ID is based on the current machine name.
+        /// </summary>
+        public string HostInstanceId
+        {
+            get
+            {
+                if (_hostInstanceId == null)
+                {
+                    _hostInstanceId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X", CultureInfo.InvariantCulture).PadLeft(32, '0');
+
+                    _hostInstanceId = _hostInstanceId.Substring(0, 32).ToLowerInvariant();
+                }
+
+                return _hostInstanceId;
+            }
+
```

Comment:
This logic was all taken from Script -- we should be able to remove that and use this property instead.